src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -84,35 +116,57 @@ const addInvoiceForRecipient = async ({
         .withDescription({ description: memo, descriptionHash })
         .generatedForRecipient()
         .withRecipientWallet(recipientWalletDescriptor)
+        .withExpiration(expiresIn)
         .withAmount(amount),
   })
 
 export const addInvoiceForRecipientForBtcWallet = async (
-  args: AddInvoiceForRecipientArgs,
+  args: AddInvoiceForRecipientForBtcWalletArgs,
 ): Promise<LnInvoice | ApplicationError> => {
-  const walletIdChecked = checkedToWalletId(args.recipientWalletId)
-  if (walletIdChecked instanceof Error) return walletIdChecked
+  const recipientWalletId = checkedToWalletId(args.recipientWalletId)
+  if (recipientWalletId instanceof Error) return recipientWalletId
 
-  const validated = await validateIsBtcWallet(walletIdChecked)
-  return validated instanceof Error ? validated : addInvoiceForRecipient(args)
+  const expiresIn = checkedToMinutes(args.expiresIn || defaultBtcExpiration)
+  if (expiresIn instanceof Error) return expiresIn
+
+  const validated = await validateIsBtcWallet(recipientWalletId)
+  if (validated instanceof Error) return validated
+  return addInvoiceForRecipient({ ...args, recipientWalletId, expiresIn })
 }
 
 export const addInvoiceForRecipientForUsdWallet = async (
-  args: AddInvoiceForRecipientArgs,
+  args: AddInvoiceForRecipientForUsdWalletArgs,
 ): Promise<LnInvoice | ApplicationError> => {
-  const walletIdChecked = checkedToWalletId(args.recipientWalletId)
-  if (walletIdChecked instanceof Error) return walletIdChecked
+  const recipientWalletId = checkedToWalletId(args.recipientWalletId)
+  if (recipientWalletId instanceof Error) return recipientWalletId
 
-  const validated = await validateIsUsdWallet(walletIdChecked)
-  return validated instanceof Error ? validated : addInvoiceForRecipient(args)
+  const expiresIn = checkedToMinutes(args.expiresIn || defaultUsdExpiration)
+  if (expiresIn instanceof Error) return expiresIn
+
+  const validated = await validateIsUsdWallet(recipientWalletId)
+  if (validated instanceof Error) return validated
+  return addInvoiceForRecipient({ ...args, recipientWalletId, expiresIn })
 }
 
 export const addInvoiceNoAmountForRecipient = async ({
   recipientWalletId,
   memo = "",
-}: AddInvoiceNoAmountForRecipientArgs): Promise<LnInvoice | ApplicationError> =>
-  addInvoice({
-    walletId: recipientWalletId,
+  expiresIn,
+}: AddInvoiceNoAmountForRecipientArgs): Promise<LnInvoice | ApplicationError> => {
+  const walletId = checkedToWalletId(recipientWalletId)
+  if (walletId instanceof Error) return walletId
+
+  let defaultExpiresIn = defaultBtcExpiration
+  const validated = await validateIsBtcWallet(walletId)
+  if (validated instanceof Error) {
+    defaultExpiresIn = defaultUsdExpiration
+  }
+
+  const expiresInChecked = checkedToMinutes(expiresIn || defaultExpiresIn)
```

Comment:
we should not allow longer expiration time for USD invoice. this opens an attack vector by enabling longer time frame thus making easy to arbitrage the trade.

Reply:
yes, this is covered https://github.com/GaloyMoney/galoy/pull/2675/files#diff-8c5feb755004a1236e79796941161c08424552caf5fb730112c429a5b513d3a4R30